src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicListener.cs

```diff
@@ -7,16 +7,20 @@
 using System.Net.Quic.Implementations.MsQuic.Internal;
 using System.Net.Security;
 using System.Security.Cryptography.X509Certificates;
+using System.Reflection;
 using System.Runtime.InteropServices;
 using System.Threading;
 using System.Threading.Channels;
 using System.Threading.Tasks;
 using static System.Net.Quic.Implementations.MsQuic.Internal.MsQuicNativeMethods;
+using System.Security.Authentication;
 
 namespace System.Net.Quic.Implementations.MsQuic
 {
     internal sealed class MsQuicListener : QuicListenerProvider, IDisposable
     {
+        private static readonly ConstructorInfo _SslClientHelloInfo = typeof(SslClientHelloInfo).GetConstructors(BindingFlags.Instance | BindingFlags.NonPublic)![0];
```

Comment:
Why can't this be made public?

If this feature is important enough to do for 6.0, then it seems like it's probably important enough to do it properly -- meaning, make the appropriate public changes to System.Net.Security etc to make it work.

Reply:
This generally falls to the same boat as `ServerCertificateContext` tracked by #53507.
To me the important part was to make HTTP/3 transition easier e.g. to preserve existing behavior.
I think we can still ask for API changes but we are slowly running out of time.